src/Microsoft.Sbom.Api/Output/Telemetry/TelemetryRecorder.cs

```diff
@@ -194,6 +195,15 @@ public void RecordException(Exception exception)
         this.exceptions.Add(exception);
     }
 
+    /// <summary>
+    /// Record the total number of packages that were processed during the execution of the SBOM tool.
+    /// </summary>
+    /// <param name="packageCount">The total package count after execution.</param>
+    public void RecordTotalNumberOfPackages(int packageCount)
+    {
+        Interlocked.Exchange(ref this.totalNumberOfPackages, packageCount);
```

Comment:
Why do you need this? aren't integer operations atomic by default?

Reply:
I'll change this to a normal operation. This was left from my previous implementation where I was counting packages in the ComponentBaseWalker which could be used by multiple threads.